### PR TITLE
kvserver: enumerate all possible paths to extract region error.

### PR DESCRIPTION
**Pull request by disksing**
@ngaut @morefreeze @BusyJay PTAL

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM

File: src/server/kv.rs [hunk 1]
```diff
@@ -286,8 +286,10 @@ impl StoreHandler {
 
 fn extract_region_error<T>(res: &StorageResult<T>) -> Option<RegionError> {
     match *res {
-        Err(StorageError::Txn(TxnError::Engine(EngineError::Request(ref err)))) => {
-            Some(err.to_owned())
+        Err(StorageError::Engine(EngineError::Request(ref e))) |
```
Comment (BusyJay) [inline src/server/kv.rs]: Maybe we can just use `Error::cause` to search recursively.

Comment (siddontang) [inline src/server/kv.rs]: +1

the error path is too long and hierarchy is too deep.

Comment (disksing) [inline src/server/kv.rs]: @BusyJay @morefreeze @siddontang 
Bad news. The `Error::cause` does not follow the error chain but only looks up for 1 level. Seems insufficient in our case.

**Comment by BusyJay** [comment]
LGTM

**Comment by morefreeze** [comment]
LGTM

**Pull request merged**